cdap-elastic/src/main/java/io/cdap/cdap/metadata/elastic/ElasticsearchMetadataStorage.java

```diff
@@ -1084,10 +1089,63 @@ private QueryBuilder createTermQuery(QueryTerm queryTerm, String textField, Sear
         : createTermQuery(queryTerm, NESTED_VALUE_FIELD, term));
     QueryBuilder propertyQuery = QueryBuilders.nestedQuery(PROPS_FIELD, boolQuery, ScoreMode.Max);
 
+    if (queryTerm.getSearchType() == QueryTerm.SearchType.DATE) {
+      BoolQueryBuilder dateQuery = createDateQuery(queryTerm, field, request);
+      return new BoolQueryBuilder().should(plainQuery).should(propertyQuery).should(dateQuery).minimumShouldMatch(1);
+    }
+
     // match either a plain term of the form "f:t" or the word "t" in property "f"
     return new BoolQueryBuilder().should(plainQuery).should(propertyQuery).minimumShouldMatch(1);
   }
 
+  /**
+   * Creates a bool query for a date search. If there is no field, creation times and user defined dates are queried.
+   * Otherwise, a field refers to a user defined property, which is searched for with a nested query.
+   *
+   * @param queryTerm QueryTerm object containing the term and its search properties
+   * @param field property field to search in. If undefined search all properties and creation times
+   * @param request the search request
+   * @return a structured query for a date search
+   */
+  private BoolQueryBuilder createDateQuery(QueryTerm queryTerm, @Nullable String field, SearchRequest request) {
+    if (field == null) {
+      QueryBuilder creationDateQuery = createDateRangeQuery(queryTerm, "created").boost(2);
+      QueryBuilder nestedDateQuery = createDateRangeQuery(queryTerm, NESTED_DATE_FIELD);
+      QueryBuilder usersDateQuery = QueryBuilders.nestedQuery(PROPS_FIELD, nestedDateQuery, ScoreMode.Avg);
+      return new BoolQueryBuilder().should(creationDateQuery).should(usersDateQuery).minimumShouldMatch(1);
+    }
+    BoolQueryBuilder dateBoolQuery = buildNestedQuery(field, request);
+    dateBoolQuery.must(createDateRangeQuery(queryTerm, NESTED_DATE_FIELD));
+    QueryBuilder propertyDateQuery = QueryBuilders.nestedQuery(PROPS_FIELD, dateBoolQuery, ScoreMode.Max);
+    return new BoolQueryBuilder().should(propertyDateQuery);
+  }
+
+  /**
+   * Creates a range query for a date search.
+   *
+   * @param queryTerm    QueryTerm object containing the term and its search properties
+   * @param field        property field to search in
+   * @return a structured range query as a QueryBuilder
+   */
+  private QueryBuilder createDateRangeQuery(QueryTerm queryTerm, String field) {
+    Long date = queryTerm.getDate();
+    QueryTerm.Comparison comp = queryTerm.getComparison();
+
+    if (comp.equals(QueryTerm.Comparison.GREATER)) {
```

Comment:
lets use switch case similar to `createTermQuery` method

Reply:
ok